newIDE/app/src/VariablesList/index.js

```diff
@@ -149,22 +174,40 @@ export default class VariablesList extends React.Component<Props, State> {
           variable,
           depth + 1,
           index,
-          parentVariable
+          parentVariable,
+          origin
         );
       })
     );
   }
 
+  _getVariableOrigin = (name: string) => {
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
+    return inheritedVariablesContainer // We check for 3 types of variable states, when editing instance variables
```

Comment:
I wonder if a two if/else would not be more readable than 4 ternaries operators? 😬 
Usually, people start complaining when there are two, and while I'm not usually that concerned about them, I think here checks are redundants and make things look more complex than what things are. Let's keep things simple and stupid! (That's actually the hardest part in software development)

```
if (!inheritedVariablesContainer || !inheritedVariablesContainer.has(name)) return '';

return variablesContainer.has(name) ? 'inherited' : 'parent';
```